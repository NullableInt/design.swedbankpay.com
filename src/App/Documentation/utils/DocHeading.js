import React from "react";

const _generateTitle = path => {
    // TODO: overkill? [EH]
    return path.match(/[^/]*$/)[0]
        .split("-")
        .map(word => word.charAt(0).toUpperCase() + word.substr(1))
        .join(" ");
};

const DocViewHeading = ({ location }) => {
    const title = _generateTitle(location.pathname);

    document.title = `${title} | PayEx DesignGuide`;
    return (
        <div className="view-heading">
            <h1>{title}</h1>
        </div>
    );
};

export default DocViewHeading;
