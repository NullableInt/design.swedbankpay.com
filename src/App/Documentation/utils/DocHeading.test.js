import React from "react";
import { shallow } from "enzyme";

import DocHeading from "./DocHeading";

describe("Utilities: DocHeading", () => {
    it("is defined", () => {
        expect(DocHeading).toBeDefined();
    });

    it("renders heading and updates document title", () => {
        const location = { pathname: "test" };

        const wrapper = shallow(
            <DocHeading location={location} />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<h1>Test</h1>)).toEqual(true);
        expect(document.title).toEqual("Test | PayEx DesignGuide");
    });

    it("correctly formats title", () => {
        const location = { pathname: "test-test" };

        const wrapper = shallow(
            <DocHeading location={location} />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<h1>Test test</h1>)).toEqual(true);
        expect(document.title).toEqual("Test test | PayEx DesignGuide");
    });
});

