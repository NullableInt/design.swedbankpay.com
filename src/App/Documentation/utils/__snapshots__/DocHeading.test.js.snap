// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: DocHeading correctly formats title 1`] = `
<div
  className="view-heading"
>
  <h1>
    Test Test
  </h1>
</div>
`;

exports[`Utilities: DocHeading renders heading and updates document title 1`] = `
<div
  className="view-heading"
>
  <h1>
    Test
  </h1>
</div>
`;
